libs/domain/product/src/services/adapter/normalizers/facet-rating/facet-rating.normalizer.spec.ts

```diff
@@ -0,0 +1,162 @@
+import { createInjector, destroyInjector } from '@spryker-oryx/di';
+import { ApiProductListModel } from '@spryker-oryx/product';
+import { RouterService } from '@spryker-oryx/router';
+import { of, take } from 'rxjs';
+import { beforeEach } from 'vitest';
+import { facetRatingNormalizer } from './facet-rating.normalizer';
+
+const mockRatingFacet: ApiProductListModel.RangeFacet = {
+  activeMax: 5,
+  activeMin: 4,
+  config: { parameterName: 'rating[min]', isMultiValued: false },
+  docCount: null,
+  localizedName: 'Product rating',
+  max: 5,
+  min: 1,
+  name: 'Rating',
+};
+
+class MockRouterService implements Partial<RouterService> {
+  currentQuery = vi.fn().mockReturnValue(of({}));
+}
+
+const generateNoramalizedFacet = (
+  facet: ApiProductListModel.RangeFacet,
+  disabledValues?: number[],
+  selectedValue?: number
+) => {
+  return {
+    name: facet.localizedName,
+    parameter: facet.config.parameterName,
+    selectedValues: selectedValue ? [String(selectedValue)] : [],
+    valuesTreeLength: facet.max - facet.min + 1,
+    values: Array.from(new Array(5).keys())
+      .reverse()
+      .map((i) => {
+        const value = i + 1;
+        return {
+          value: String(value),
+          selected: selectedValue ? selectedValue === i + 1 : false,
+          count: 0,
+          disabled: disabledValues ? disabledValues.includes(i + 1) : false,
+        };
+      }),
+  };
+};
+
+describe('Product Facet Normalizers', () => {
+  let routerService: MockRouterService;
+
+  beforeEach(() => {
+    const injector = createInjector({
+      providers: [
+        {
+          provide: RouterService,
+          useClass: MockRouterService,
+        },
+      ],
+    });
+
+    routerService = injector.inject<MockRouterService>(RouterService);
+  });
+
+  afterEach(() => {
+    destroyInjector();
+    vi.clearAllMocks();
+  });
+
+  const callback = vi.fn();
+
+  it('should return normalized rating facet-navigation', () => {
+    facetRatingNormalizer(mockRatingFacet).pipe(take(1)).subscribe(callback);
```

Comment:
Try to keep the same structure. Some of your tests subscribing inside `beforeEach`, some in the `it`